modules/tinymce/src/core/test/ts/browser/content/EditorContentTest.ts

```diff
@@ -17,16 +18,66 @@ const defaultExpectedEvents = [
 ];
 
 describe('browser.tinymce.core.content.EditorContentTest', () => {
+  const toHtml = (node: AstNode): string => {
+    const htmlSerializer = HtmlSerializer({});
+    return htmlSerializer.serialize(node);
+  };
+
+  const assertContentTreeEqualToHtml = (editor: Editor, html: string, msg: string) => {
+    const tree = editor.getContent({ format: 'tree' });
+    assert.equal(toHtml(tree), html, msg);
+  };
+
+  const getFontTree = (): AstNode => {
+    const body = new AstNode('body', 11);
+    const font = new AstNode('font', 1);
+    const text = new AstNode('#text', 3);
+
+    text.value = 'x';
+    font.attr('size', '7');
+    font.append(text);
+    body.append(font);
+
+    return body;
+  };
+
+  const testSetContentTreeWithContentAlteredInBeforeSetContent = (editor: Editor, alteredContent: string, msg: string, expectedContent?: string) => {
+    editor.setContent('<p>tree</p>');
+    editor.once('BeforeSetContent', (e) => {
+      assert.equal(e.content, '<font size="7">x</font>');
+      e.content = alteredContent;
+    });
+    editor.setContent(getFontTree());
+    assert.equal(editor.getContent(), expectedContent ?? alteredContent, msg);
+  };
+
+  const testGetContentTreeWithContentAlteredInGetContent = (editor: Editor, alteredContent: string, msg: string, expectedContent?: string) => {
+    editor.setContent('<p>tree</p>');
+    editor.once('GetContent', (e) => {
+      assert.equal(e.content, '<p>tree</p>');
+      e.content = alteredContent;
+    });
+    assertContentTreeEqualToHtml(editor, expectedContent ?? alteredContent, msg);
+  };
+
   Arr.each(
     [
       {
         inline: true
       },
       {
         inline: false
+      },
+      {
+        inline: true,
+        xss_sanitization: false
+      },
+      {
+        inline: false,
+        xss_sanitization: false
       }
     ], (options) => {
-      context('Test with inline option ' + options.inline, () => {
+      context('Test with inline: ' + options.inline + ' and xss_sanitization: ' + options.xss_sanitization, () => {
```

Comment:
Maybe convert to template string while you are at it? But feel free to skip it since it's mostly code style.

```suggestion
      context(`Test with inline: ${options.inline} and xss_sanitization: ${options.xss_sanitization}`, () => {
```

Reply:
Converted all concatenated strings to template strings.